internal/cmd/skupper/listener/listener_create.go

```diff
@@ -1,47 +1,188 @@
+	listenerCreateExample = "skupper listener create database 5432"
+
+	listenerTypes = []string{"tcp", "http"}
 )
 
+type ListenerFlags struct {
+	routingKey   string
+	host         string
+	tlsSecret    string
+	listenerType string
+}
+
 type CmdListenerCreate struct {
-	client   *client.VanClient
-	CobraCmd cobra.Command
+	client    skupperv1alpha1.SkupperV1alpha1Interface
+	CobraCmd  cobra.Command
+	flags     ListenerFlags
+	namespace string
+	name      string
+	port      int
 }
 
 func NewCmdListenerCreate() *CmdListenerCreate {
 
-	skupperCmd := CmdListenerCreate{}
+	skupperCmd := CmdListenerCreate{flags: ListenerFlags{}}
 
 	cmd := cobra.Command{
-		Use:     "create",
-		Short:   "",
+		Use:     "create <name>",
+		Short:   "create a listener",
 		Long:    listenerCreateLong,
 		Example: listenerCreateExample,
 		PreRun:  skupperCmd.NewClient,
 		Run: func(cmd *cobra.Command, args []string) {
 			utils.HandleErrorList(skupperCmd.ValidateInput(args))
-			skupperCmd.InputToOptions()
 			utils.HandleError(skupperCmd.Run())
 		},
+		PostRunE: func(cmd *cobra.Command, args []string) error {
+			return skupperCmd.WaitUntilReady()
+		},
 	}
 
 	skupperCmd.CobraCmd = cmd
+	skupperCmd.AddFlags()
 
 	return &skupperCmd
 }
 
-func (cmd *CmdListenerCreate) NewClient(cobraCommand *cobra.Command, args []string) {}
-func (cmd *CmdListenerCreate) AddFlags()                                            {}
-func (cmd *CmdListenerCreate) ValidateInput(args []string) []error                  { return nil }
-func (cmd *CmdListenerCreate) InputToOptions()                                      {}
-func (cmd *CmdListenerCreate) Run() error                                           { fmt.Println("Not implemented yet."); return nil }
-func (cmd *CmdListenerCreate) WaitUntilReady() error                                { return nil }
+func (cmd *CmdListenerCreate) NewClient(cobraCommand *cobra.Command, args []string) {
+	cli, err := client.NewClient("", "", "")
+	utils.HandleError(err)
+	cmd.client = cli.GetSkupperClient().SkupperV1alpha1()
+	cmd.namespace = cli.Namespace
+}
+
+func (cmd *CmdListenerCreate) AddFlags() {
+	cmd.CobraCmd.Flags().StringVarP(&cmd.flags.routingKey, "routing-key", "r", "", "The identifier used to route traffic from listeners to connectors")
+	cmd.CobraCmd.Flags().StringVar(&cmd.flags.host, "host", "", "The hostname or IP address of the local listener")
+	cmd.CobraCmd.Flags().StringVar(&cmd.flags.tlsSecret, "tls-secret", "", "The name of a Kubernetes secret containing TLS credentials")
+	cmd.CobraCmd.Flags().StringVarP(&cmd.flags.listenerType, "type", "t", "tcp", "The listener type. Choices: [tcp|http].")
+}
+
+func (cmd *CmdListenerCreate) ValidateInput(args []string) []error {
+	var validationErrors []error
+	resourceStringValidator := validator.NewResourceStringValidator()
+	tlsStringValidator := validator.NewTlsStringValidator()
```

Comment:
Because the secret is the name of an existent kubernetes resource, in this case a secret, I would remove the tlsStringValidator and check if the secret exists in the namespace. To do this, the  CmdListenerCreate will need to have a kubeClient

Reply:
I've created the kubeClient,  but am having trouble figuring out how to create a fake kubeClient for the unit test.

Reply:
done, thanks Neo